runners/core-java/src/main/java/org/apache/beam/runners/core/TimerInternals.java

```diff
@@ -201,8 +201,9 @@ public static TimerData of(StateNamespace namespace, Instant timestamp, TimeDoma
     /**
      * {@inheritDoc}.
      *
-     * <p>The ordering of {@link TimerData} that are not in the same namespace or domain is
-     * arbitrary.
+     * <p>Compares {@link TimerData} by using its fields in this order: timestamp, domain,
```

Comment:
I probably was wrong when I wrote the JIRA - the big reason to compare with all of the fields is because otherwise the comparator is inconsistent with equals (two timers in the same domain and namespace with the same timestamp compare as equal, even with different IDs, and that's very dangerous within any ordered data structure). It is generally important that we order timers across namespaces and domains by timestamp, and then domains and namespaces provide an arbitrary sub-order

Reply:
Are you suggesting me to change the javadoc? or just mentioning?

Reply:
Yeah - specifically, this is for consistency with equals within a single JVM, not consistent across different JVMs or versions